src/backend/utils/adt/pg_locale.c

```diff
@@ -1534,6 +1537,11 @@ pg_newlocale_from_collation(Oid collid)
 	}
 
 	cache_entry = lookup_collation_cache(collid, false);
+    
+	if (prev_cache && prev_cache->collid == collid)
```

Comment:
Should wrap those code into a hook, and put the impl into babel code

Reply:
Here caching the very previous call using global pointer to avoid hash search. Creating a hook for this may be unnecessary

Reply:
We should try our best to make minimal change on PG side